src/pharmpy/tools/amd/run.py

```diff
@@ -396,9 +396,8 @@ def run_amd(
         model = filter_dataset(model, 'DVID != 2')
 
     if results is None:
-        model = run_tool('modelfit', model, path=db.path / 'modelfit', resume=resume)
-    else:
-        model = model.replace(modelfit_results=results)
+        results = run_tool('modelfit', model, path=db.path / 'modelfit', resume=resume)
+    model = model.replace(modelfit_results=results)
```

Comment:
The idea is that `Model` should not contain `modelfit_results`

Reply:
I think this is temporary until `ModelEntry` is implemented in AMD? We already did this before this PR.

Reply:
Ok, so AMD is still on todo.

Reply:
Yes, it's the final tool to be implemented.